File: ProyectoSpring/SpringBussinesOne/demo/demo/src/main/java/com/BussinesOne/demo/models/Fichaje.java
package com.BussinesOne.demo.models;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

import com.fasterxml.jackson.annotation.JsonManagedReference;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name = "Fichaje")
public class Fichaje {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long fichajeID;

    @ManyToOne
    @JsonManagedReference
    @JoinColumn(name = "empleadoid")
    private Perfil empleado;

    @Column(nullable = false)
    private LocalDate fecha;

    @Column(name = "horaentrada", nullable = false)
    private LocalTime horaEntrada;

    @Column(name = "horasalida")
    private LocalTime horaSalida;

    @Column(name = "tiemporeal", nullable = false)
    private Boolean tiempoReal = true;

    @Column(name = "createdat", nullable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
    
}

File: ProyectoAndroid/app/src/main/java/com/example/proyectobussinesone/ComponenteKotlinPrueba/models/Fichaje.kt
package com.example.proyectobussinesone.ComponenteKotlinPrueba.models

import com.google.gson.annotations.SerializedName
import java.time.LocalDate
import java.time.LocalTime




data class Fichaje(
    @SerializedName("fichajeID")
    val fichajeID: Long,

    val empleado: Perfil,

    // JSON: "fecha": "2025-06-01"
    // Lo dejamos como String para que GSON no intente mapear a un objeto LocalDate
    val fecha: String,

    // JSON: "horaEntrada": "08:30:00"
    val horaEntrada: String,

    // JSON: "horaSalida": "17:00:00"  (puede ser null si no hay fichaje de salida)
    val horaSalida: String?,

    val tiempoReal: Boolean,

    // JSON: "createdAt": "2025-05-29T19:58:52.315646"
    val createdAt: String?
)